fix: poll interval to request Notion

telekosmos opened the pull request:
In order to know when the export is finished, the poll interval to Notion was set to 50ms. Notion didn't bother this one (I guess they didn't pay attention ever before) but from little time ago, on this very frequent request a 429 is being received. So, as the export use to take from seconds to minutes, was tested than a value of 5000ms is ok and don't bother Notion (for now...).

Commit:
fix: poll interval to request Notion

In order to know when the export is finished, the poll interval to Notion
was set to 50ms. Notion didn't bother this one (I guess they didn't pay
attention ever before) but from little time ago, on this very frequent
request a 429 is being received. So, as the export use to take from
seconds to minutes, was tested than a value of 5000ms is ok and
don't bother Notion (for now...).

yannbolliger commented:
Thanks for the PR. I have also seen the 429 behaviour recently. 
However, I feel like 5000 is quite a big increase. Could we set the default to 100 and make the value configurable via the `Config` ([see here](https://github.com/yannbolliger/notion-exporter/blob/master/src/config.ts#L2-L13))?

telekosmos commented:
> Thanks for the PR. I have also seen the 429 behaviour recently. However, I feel like 5000 is quite a big increase. Could we set the default to 100 and make the value configurable via the `Config` ([see here](https://github.com/yannbolliger/notion-exporter/blob/master/src/config.ts#L2-L13))?

I already tried to set it to 500 ms and didn't work (see [here](https://github.com/onebeyond/notion-exporter/commit/8c5e049f8646737a8735a782602835e56feb5249)).

You're right regarding to the `Config`. I'll amend it

Commit:
feat: poll interval in config

Add a property (optional) in the config to set the poll interval
to check if the export has finished (default to 2000ms).

yannbolliger commented on src/NotionExporter.ts:
```diff
@@ -69,15 +69,15 @@ export class NotionExporter {
   private pollTask = (
     taskId: string,
-    pollInterval: number = 50
+    pollInterval: number = 500
```
Maybe we could even remove this now, as we have it in `Config`?

telekosmos replied:
Done. Hopefully they are not bothered for the 2 secs in the future 😄

yannbolliger reviewed (commented):
Let's acutally remove the function parameter now. Otherwise 👍

Commit:
chore: poll interval in config

Removed as parameter function and make it mandatory in config.

yannbolliger reviewed (approved):
Nice, thanks for contributing.

yannbolliger commented on src/config.ts:
```diff
@@ -10,10 +10,13 @@ export interface Config {
   /** Export all blocks of the DB/page or just the ones in the current view. Default: "all" */
   collectionViewExportType?: "currentView" | "all"
+  /** Poll export task finished interval in ms */
+  pollInterval: number
```
Let's actually leave this as optional because otherwise, one always has to provide it. 
(Also it causes a TS error in the testing pipeline...)

telekosmos replied:
Sorry about [that](https://github.com/yannbolliger/notion-exporter/actions/runs/9224775012/job/25426053845?pr=9) 😞 -> I'll review (carefully) and fix it

Commit:
fix: missing poll interval value

Set poll internal as an optional value in config and added private
constant with a default interval value in the case the config one
is not provided. Be aware it is used _only_ in a private method

yannbolliger commented on src/NotionExporter.ts:
```diff
@@ -14,6 +14,7 @@ interface Task {
 export class NotionExporter {
   protected readonly client: AxiosInstance
   private readonly config: Config
+  private readonly _DEFAULT_POLL_INTERVAL = 2000
```
Why don't we use the default from `defaultConfig`? ...

yannbolliger commented on src/NotionExporter.ts:
```diff
@@ -68,22 +69,21 @@ export class NotionExporter {
           resolve(task.status.exportURL)
         else if (task.state === "in_progress" || task.state === "not_started") {
-          setTimeout(poll, pollInterval)
+          setTimeout(poll, this.config.pollInterval || this._DEFAULT_POLL_INTERVAL)
```
... that would change this to:
```suggestion
          setTimeout(poll, this.config.pollInterval || defaultConfig.pollInterval)
```
(same below)

yannbolliger committed:
Fix linting and use default config.

yannbolliger commented:
Thx for contributing! Just released this: https://github.com/yannbolliger/notion-exporter/releases/tag/v0.6.2
